### PR TITLE
[W-12474663] add coveo recommendations for the "popular topics" section

### PR DESCRIPTION
**Pull request by cheungaryk**
ref: W-12474663

The "Popular Topics" section in the landing page is currently a list of hardcoded links. This PR changes the content to be dynamically generated using Coveo's [recommendations](https://docs.coveo.com/en/atomic/latest/usage/atomic-recommendations/) feature to auto-populate this list. To make sure this list is generated correctly, we need to implement Coveo User Analytics, which was submitted in #523.

The following screenshot shows the look & feel of the dynamically generated content:

Screenshot 2023-10-19 at 12 42 44 PM

### CODE REVIEW

**Comment by cheungaryk** [comment]
Apologies for the omnibus PR. I am going to break down the changes in this PR:

- package and package-lock.json changes are from #558 which led to slight property adjustments in src/css/adoc/doc.css, src/css/components/breadcrumbs.css and src/css/components/deployment-options.css, and formatting in src/js/01-nav.js, and src/js/vendor/icondefs.js
- the header files are automatically updated by the "preview" script
- the rest are the ones that are relevant:
  -  src/css/vendor/coveo.css has the look and feel of the content recommendations list
  - src/partials/landing-page/landing-page.hbs and src/partials/landing-page/popular-topics.hbs define the HTML of the content recommendation interface
  - the end of src/partials/head/head-scripts.hbs (the changes after line 46) add the content recommendations script and upgrade Coveo script version. The rest are formatting changes
  - src/js/21-coveo-facets.js is uncertain. I am asking Clint if I want to include it. What it does is checking the latest version of the version filter facet by default, so users can only see latest version pages until the version filters are changed. Once confirmed, I will un-draft this PR

**Review by jdomingu — APPROVED** [review]
Looks good! No real comments. It's a shame that we need to include `!important` flags in the css, but I'm just going to assume you tried everything else and that was the only way to override the Coveo styling.

One suggestion that's somewhat unrelated. I noticed that the `MuleSoft Popular Topics` coveo key isn't in our group LastPass. Would it be worth adding it there? 

If you do get it added in, I'm happy to try it locally and do some more testing. LMK if that would be helpful.

**Comment by cheungaryk** [comment]
> Looks good! No real comments. It's a shame that we need to include `!important` flags in the css, but I'm just going to assume you tried everything else and that was the only way to override the Coveo styling.
> 
> One suggestion that's somewhat unrelated. I noticed that the `MuleSoft Popular Topics` coveo key isn't in our group LastPass. Would it be worth adding it there?
> 
> If you do get it added in, I'm happy to try it locally and do some more testing. LMK if that would be helpful.

- Yes, the `!important` flags were there to override Coveo settings.
- The coveo key is in LastPass, under the name "coveo prod token". Assuming that you mean the key to perform the searches, because the content recommendation key is already hardcoded in the code.

File: src/partials/landing-page/popular-topics.hbs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        <atomic-recs-interface id="recs" pipeline="MS Docs Site - Popular Topics" search-hub="Popular Topics">
+            {{!-- number-of-recommendations seems buggy. We want 7 items but can't specify 7 directly, only another number (like 10) would yield 7 results. Please test for the exact number in the preview site. --}}
+            <atomic-recs-list density="compact" heading-level=2 label="Popular Topics" number-of-recommendations=10>
+                <atomic-recs-result-template>
+                    <template>
+                        <style>
+                            atomic-result-section-title {
+                                padding-bottom: 2px;
+                            }
+                            atomic-result-link {
```
[the remark this answers is not in the document]
Comment (cheungaryk) [reply src/partials/landing-page/popular-topics.hbs:13]: added spaces as requested.

**Review by jdomingu — APPROVED** [review]
Makes sense 👍 Thanks for making those changes.

Clicked approve. I just want to confirm that this is what I should be seeing:

Screenshot 2023-10-20 at 2 04 35 PM

**Comment by jdomingu** [comment]
Hey! Sorry I didn't get to this in time for the merge. TLDR: No issue on your end.

The recs weren't showing up for me because you need _a_ coveo key defined or the coveo code doesn't load. I.e. all the coveo code is wrapped in an if-block here: 
<https://github.com/mulesoft/docs-site-ui/blob/ab401ff7b7fc5d7bf56a11c8ab2a48c861cf47c5/src/partials/head/head-scripts.hbs#L45>

Setting a dummy token like this was enough for the recs to load:
`export COVEO_API_KEY="blah"`

**Pull request merged**